File: examples/javascript/kx122.js
/*
* Author: Noel Eck <noel.eck@intel.com>
* Copyright (c) 2017 Intel Corporation.
*
* This program and the accompanying materials are made available under the
* terms of the The MIT License which is available at
* https://opensource.org/licenses/MIT.
*
* SPDX-License-Identifier: MIT
*/

var kx122 = require("jsupm_kx122");

// Instantiate a KX122
var kx122_sensor = new kx122.KX122(0, -1, 24);

// Reset the device
kx122_sensor.softwareReset();

// Reset the device
kx122_sensor.deviceInit(kx122.KX122_ODR_50,
		kx122.HIGH_RES,
		kx122.KX122_RANGE_2G);

var myInterval = setInterval(function()
{
	data = kx122_sensor.getAccelerationDataVector();
	console.log("Acceleration x: " + data.get(0)
		+ " y: " + data.get(1)
		+ " z: " + data.get(2));
}, 1000);

// When exiting: clear interval and print message
process.on('SIGINT', function()
{
	clearInterval(myInterval);
	console.log("Exiting...");
	process.exit(0);
});

File: examples/c/kx122.c
/*
* The MIT License (MIT)
*
* Author: Samuli Rissanen <samuli.rissanen@hotmail.com>
* Copyright (c) 2018 Rohm Semiconductor.
*
* This program and the accompanying materials are made available under the
* terms of the The MIT License which is available at
* https://opensource.org/licenses/MIT.
*
* SPDX-License-Identifier: MIT
*/

#include <stdio.h>
#include <unistd.h>
#include <signal.h>

#include "kx122.h"
#include "upm_utilities.h"

bool shouldRun = true;

void sig_handler(int signo)
{
  if(signo == SIGINT){
    shouldRun = false;
  }
}

int main(int argc, char **argv)
{
  signal(SIGINT,sig_handler);

  //! [Interesting]
  kx122_context sensor = kx122_init(0, -1, 24, 10000);
  if (!sensor)
  {
      printf("kx122_init() failed.\n");
      return 1;
  }

  kx122_sensor_software_reset(sensor);
  kx122_device_init(sensor,KX122_ODR_50,HIGH_RES,KX122_RANGE_2G);

  float x,y,z;
  int wait_time = (kx122_get_sample_period(sensor) * MICRO_S);
  if (wait_time < 0) wait_time = 1000;

  while(shouldRun){
    kx122_get_acceleration_data(sensor,&x,&y,&z);

    printf("%.02f | %.02f | %.02f\n",x,y,z);
    upm_delay_us(wait_time);
  }

  kx122_close(sensor);
  //! [Interesting]

  return 0;
}
